spot-client/src/common/remote-control/BaseRemoteControlService.js

```diff
@@ -206,9 +207,13 @@ export class BaseRemoteControlService extends Emitter {
      * @returns {void}
      */
     _onDisconnect(reason) {
-        this.disconnect()
-            .then(() => this.emit(
-                SERVICE_UPDATES.UNRECOVERABLE_DISCONNECT, reason));
+        // If XMPP server is not working calling disconnect triggers onDisconnect again.
+        if (!this._disconnecting) {
+            this._disconnecting = true;
+            this.disconnect()
+                .then(() => this.emit(
+                    SERVICE_UPDATES.UNRECOVERABLE_DISCONNECT, reason));
+        }
```

Comment:
Should disconnecting be set to false after disconnecting is finished?

Reply:
I don't think so. Once disconnect has been called it is not safe to call it again until the service is connected again.